dashbuilder/dashbuilder-shared/dashbuilder-dsl/src/main/java/org/dashbuilder/dsl/serialization/impl/InternalDataSetProviderRegistry.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dashbuilder.dsl.serialization.impl;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.dashbuilder.dataprovider.DataSetProvider;
+import org.dashbuilder.dataprovider.DataSetProviderRegistry;
+import org.dashbuilder.dataprovider.DataSetProviderType;
+import org.dashbuilder.kieserver.RuntimeKieServerDataSetProviderType;
+
+public class InternalDataSetProviderRegistry implements DataSetProviderRegistry {
+
+    Set<DataSetProviderType> providers;
+
+    public InternalDataSetProviderRegistry() {
+        initProviders();
+    }
+
+    @Override
+    public void registerDataProvider(DataSetProvider dataProvider) {
+        // empty
+    }
+
+    @Override
+    public DataSetProvider getDataSetProvider(DataSetProviderType type) {
+        return null;
```

Comment:
Yes, this registry will not be building providers because to build a provider I would have to add all the provider specific Maven dependency, but I only wanted definitions for the API, the provider will the runtime server. The reason I had to create this class is because at some point the API consults it to check if a provider is valid, the only required method is `getProviderTypeByName`. Without it we have an unexpected exception. I added a comment to clarify why we return null;